apitest/views.py

```diff
@@ -0,0 +1,104 @@
+                        break
+                    record = {
+                        'key': key,
+                        'value': value
+                    }
+                    if APITestHeaderSerializer(data=record).is_valid():
+                        monitor_data['headers'][key] = value
+                    else:
+                        error_log += ["Please make sure your [headers] key and value are valid strings!"]
+                        break                    
+
+            if request.data.get('body_type') == 'FORM':
+                for key_value_pair in request.data.get('body_form'):
+                    if 'key' in key_value_pair and 'value' in key_value_pair:
+                        key, value = key_value_pair['key'], key_value_pair['value']
+                    else:
+                        error_log += ["Please make sure you submit correct [body form]"]
+                        break
+                    record = {
+                        'key': key,
+                        'value': value
+                    }
+                    if APITestBodyFormSerializer(data=record).is_valid():
+                        monitor_data['body'][key] = value
+                    else:
+                        error_log += ["Please make sure your [body form] key and value are valid strings!"]
+                        break
+            elif request.data.get('body_type') == 'RAW':
+                monitor_data['body']=request.data["raw_body"]
+
+            assert len(error_log) == 0, error_log
+
+            resp = None
+        
+            try:
+                print(f"""
+                    url = {monitor_data['url']}
+                    params = {monitor_data['query_params']}
+                    headers = {monitor_data['headers']}
+                    data = {monitor_data['body']}
+                """)
+                if monitor_data['method'] == 'GET':
+                    resp = requests.get(monitor_data['url'], params=monitor_data['query_params'], headers=monitor_data['headers'], timeout=30)
+                elif monitor_data['method'] == 'POST':
+                    resp = requests.post(monitor_data['url'], params=monitor_data['query_params'], data=monitor_data['body'], headers=monitor_data['headers'], timeout=30)
+                elif monitor_data['method'] == 'PATCH':
+                    resp = requests.patch(monitor_data['url'], params=monitor_data['query_params'], data=monitor_data['body'], headers=monitor_data['headers'], timeout=30)
+                elif monitor_data['method'] == 'PUT':
+                    resp = requests.put(monitor_data['url'], params=monitor_data['query_params'], data=monitor_data['body'], headers=monitor_data['headers'], timeout=30)
+                elif monitor_data['method'] == 'DELETE':
+                    resp = requests.delete(monitor_data['url'], params=monitor_data['query_params'], data=monitor_data['body'], headers=monitor_data['headers'], timeout=30)
```

Comment:
Ini langsung pake generic function aja dari requestnya. Jadi cukup:
```suggestion
               resp = requests.get(monitor_data['method'], monitor_data['url'], params=monitor_data['query_params'], headers=monitor_data['headers'], timeout=30)`
```

Jdi line 74 yg `resp = None` dihapus aja juga

reference:
https://requests.readthedocs.io/en/latest/api/#requests.request